modules/ROOT/pages/_partials/sync/js-client-files.adoc

```diff
@@ -0,0 +1,82 @@
+= File Upload Client
+
+{sync-client} provides support for uploading binary data along with the GraphQL queries.
+Implementation is relying on `apollo-upload-client` package build by Apollo community.
+
+== Introduction
+
+The upload functionality using the GraphQL multipart form requests specification.
+The File Upload needs to be implemented on both server and client:
+
+1) On the client HTML FileList objects are mapped into a mutation and sent to the server in a multipart request.
+
+2) On the server: The multipart request is handled. The server processes it and provides an upload argument to a resolver. 
+In the resolver function, the upload promise resolves an object.
+
+NOTE: File upload is based on link:https://github.com/jaydenseric/graphql-multipart-request-spec[graphql-multipart-request-spec]
+
+== Enabling File Upload 
+
+File upload feature needs to be enabled by passing `fileUpload` flag to config object:
+
+[source, javascript]
+----
+
+const config = {
+  ...
+  fileUpload: true
+  ...
+};
+
+//create a new client
+----
+
+= Uploading files from GraphQL 
+
+File upload capability adds new GraphQL scalar `Upload` that can be used for mutations that operate on binary data.
+`Upload` scalar maps html `FileList` HTML5 object in GraphQL schemas. 
+The first step required to work with binary uploads is to write mutation that will contain `Upload` scalar:
+
```

Comment:
The following example demonstrates how to upload a profile picture for the showcase app?